### PR TITLE
Pass `pendingProps` as argument in `createWorkInProgress`

### PR DESCRIPTION
**Pull request by zombieJ**
Update `createWorkInProgress` in `ReactFiber.js` to accept `pendingProps` as argument.

To avoid merge conflict, open new pr:
ref: https://github.com/facebook/react/pull/10643

### CODE REVIEW

**Comment by zombieJ** [comment]
@acdlite @gaearon , Rebase master & fix the conflict

File: packages/react-reconciler/src/ReactChildFiber.js [hunk 1]
```diff
@@ -288,11 +288,15 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     return existingChildren;
   }
 
-  function useFiber(fiber: Fiber, expirationTime: ExpirationTime): Fiber {
+  function useFiber(
+    fiber: Fiber,
+    expirationTime: ExpirationTime,
+    pendingProps: any,
```
Comment (acdlite) [inline packages/react-reconciler/src/ReactChildFiber.js]: Can we change this to `mixed` instead of `any`?

**Comment by zombieJ** [comment]
thx for advise. Let me update 😄

**Comment by zombieJ** [comment]
hi @acdlite ,
updated, pls check

**Comment by blling** [comment]
So many changes? 
A new PR is better maybe?

**Comment by gaearon** [comment]
Let's get this in now that the review was addressed. Thanks again!

File: packages/react-reconciler/src/ReactFiber.js [hunk 2]
```diff
@@ -4,7 +4,6 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  *
- * @flow
```
Comment (gaearon) [inline packages/react-reconciler/src/ReactFiber.js]: It's my fault for missing this in the review, but this completely removed flow typing from the most important module in the codebase.

Comment (abhaynikam) [reply packages/react-reconciler/src/ReactFiber.js]: @gaearon : Can I send a PR for adding flow typing here again?

Comment (aweary) [reply packages/react-reconciler/src/ReactFiber.js]: @gaearon for reference there's an existing PR https://github.com/facebook/react/pull/12764

**Pull request merged**